### DISCUSSION TITLE
快速安装（即提供服务器预设整合包）

### DISCUSSION THREAD

**Discussion by YaoSiQian** [MCSLTeam/MCSL2#114] [功能建议]
# 起因
TL;DR 懒癌发作
> 啊啊啊一个一个插件好麻烦啊

> TMD又要下cache，mojang我*你**

> 配插件 は 实在 無理です

> ~EULA好麻烦~
# 配置后可以解决的问题
> 所有内容均取自[MCSL2.2正式版 | 从心出发，不断向前](https://www.bilibili.com/video/BV1rN411z7no)
![Carpet](https://github.com/MCSLTeam/MCSL2/assets/29835749/0dafefe7-d898-4230-8b6c-2a31608c5c6f "Carpet")
![Server Setup](https://github.com/MCSLTeam/MCSL2/assets/29835749/9ef39ad4-d214-414d-8b50-dad0e60a7c91 "Server Setup")
![FM Limit](https://github.com/MCSLTeam/MCSL2/assets/29835749/ddac506f-11c2-47fb-b259-1f7ffb6db76c "FM Limit")
![Core Error](https://github.com/MCSLTeam/MCSL2/assets/29835749/2bb78f64-6dbe-4ef6-9d2c-007dd7bda6f6 "Core Error")
![WhynoVelocity](https://github.com/MCSLTeam/MCSL2/assets/29835749/14d93e9f-3d2d-4753-8ae3-7a7a6e1e6369 "WhynoVelocity")

# 解决方案
## 参考对象
[![MCSManager](https://github.com/MCSLTeam/MCSL2/assets/29835749/f0ad6e14-953a-4eff-89ed-54e6e25b3d03 "MCSManager")](https://github.com/MCSManager/MCSManager/blob/ab42daba1b7e3a72cede684d107e8c161f4062cb/src/app/entity/setting.ts#L36)

## 详细分析
1. 提供一份JSON，列出可以使用的服务端
```json
[
    {
        "info": "[整合包] 小型服务器推荐，包含基础插件、箱子锁、操作回滚、领地和商店等插件。",
        "mc": "Minecraft 1.19.2 Java 版整合包",
        "java": "17",
        "size": 210,
        "remark": "支持插件，最低 6GB 内存要求",
        "author": "Zhang_San",
        "targetLink": "https://example.com/mc/paper1.19.2"
    },
    {
        "info": "[原版生存] 适合朋友联机，快速开服，服务器搭建，回味老版本的地穴探索。",
        "mc": "Minecraft 17.1 Java 原版",
        "java": "17",
        "size": 160,
        "remark": "支持插件，最低 4GB 内存要求",
        "author": "Li_Si",
        "targetLink": "http://cos.example.com.cn/d/Minecraft-Server-Software/paper-1.17.1.zip"
    }
]
```
2. 根据`targetLink`下载文件，并且根据JSON中的`java`、`size`以及压缩包中的`config.json`配置服务端
```json
{
  "type": "minecraft/java",
  "startCommand": "{{java}} -jar server.jar -nogui",
  "stopCommand": "stop",
  "ie": "utf-8",
  "oe": "utf-8"
}
```
3. _**麦块，启动！**_
```log
[11:45:14 INFO]: Done (1.919s)! For help, type "help"
```

**Comment by YaoSiQian** [comment]
【喜报】本 Discussion 喜提 #114

**Comment by kitUIN** [comment]
TL;DR 
一般来说,这些整合包都是提供的百度云下载地址.....
除非咱们自己搞个平台,或者开偷😋
